internal/ingress/annotations/globalratelimit/main.go

```diff
@@ -101,11 +133,16 @@ func (a globalratelimit) Parse(ing *networking.Ingress) (interface{}, error) {
 		return nil, err
 	}
 
+	rawHeaderBasedRateLimits, _ := parser.GetStringAnnotation("global-rate-limit-header-based", ing)
+	var headerBaseRateLimits []HeaderBasedRateLimit
+	json.Unmarshal([]byte(rawHeaderBasedRateLimits), &headerBaseRateLimits)
```

Comment:
We still need to capture the error returned by `json.Unmarshal`.

Reply:
thanks, I will add it and fix the lua errors